src/sentry/static/sentry/app/components/tooltip.tsx

```diff
@@ -162,35 +171,64 @@ class Tooltip extends React.Component<Props, State> {
   };
 
   handleOpen = () => {
-    const {delay, isHoverable} = this.props;
+    const {delay} = this.props;
 
     if (this.delayHideTimeout) {
       window.clearTimeout(this.delayHideTimeout);
       this.delayHideTimeout = null;
     }
 
-    if (delay || isHoverable) {
+    if (delay || this.isHoverable()) {
       this.delayTimeout = window.setTimeout(this.setOpen, delay || IS_HOVERABLE_DELAY);
     } else {
       this.setOpen();
     }
   };
 
   handleClose = () => {
-    const {isHoverable} = this.props;
-
     if (this.delayTimeout) {
       window.clearTimeout(this.delayTimeout);
       this.delayTimeout = null;
     }
 
-    if (isHoverable) {
+    if (this.isHoverable()) {
       this.delayHideTimeout = window.setTimeout(this.setClose, IS_HOVERABLE_DELAY);
     } else {
       this.setClose();
     }
   };
 
+  isHoverable() {
+    const {variant} = this.props;
+    const isCopyable = variant === 'copyable';
+    return variant === 'hoverable' || isCopyable;
+  }
+
+  getTooltipContent() {
+    const {variant, title} = this.props;
+
+    if (variant !== 'copyable') {
+      return title;
+    }
+
+    // if variant equals copyable, the title has to be of type string,
+    // but as the compiler does not process this info correctly a parser is needed
+    return (
+      <TooltipClipboardWrapper
+        onClick={event => {
+          event.stopPropagation();
+        }}
+      >
+        <TextOverflow>{title}</TextOverflow>
+        <Clipboard value={String(title)}>
+          <TooltipClipboardIconWrapper>
+            <IconCopy size="xs" color="white" />
+          </TooltipClipboardIconWrapper>
+        </Clipboard>
+      </TooltipClipboardWrapper>
+    );
+  }
```

Comment:
Hmm. Should tooltips really be this smart, to have copy functionality?

I wonder if it would be better to compose tooltip into a new component, maybe called `ClipboardTooltip`?

Reply:
I would think having a composed component would be pretty short, and encapsulate all of this logic in that component.

It would also make the typing simpler I believe.

Reply:
I thought about that too, but as we discussed yesterday considering variants instead of creating new components I was not sure

Reply:
Had some discussion in Slack with @priscilawebdev about this.

We concluded that adding a functionality as a 'variant' like this (versus something like, having a variant prop on a loading component) is a bit of a slippery slope. Since copy functionality is a bit removed from what a tooltip typically does, it would be easy to add more functional variants, and get into a place where it becomes hard to maintain this component.

Some snippets from the convo:

> I think maybe the distinction to think about is scope of functionality. A loading indicator with a few visual variants likely makes sense, since they all “do” the same thing, but act visually different.
>
> In the case of the tooltip, I think having it “learn” to handle copying it’s content falls more towards functionality creep. I would say probably 90% of our tooltips wouldn’t use that functionality, so it might be a bit too specialized.
>
> The other thing to think about is maintainability of the component, if we add too much functionality, then it may grow too much in scope and when you want to make a change to the Tooltip, you have to spend some time understanding the other bits of functionality, even if it’s only rarely used

Just to get a 2nd opinion here though, @billyvg would you agree or disagree that this type of functionality would be better to live in something like a `ClipboardTooltip` component?

Reply:
I agree. I think props to handle variants works well when it changes things visually, but makes more sense as a different component when introducing new behaviors.